umbraco/Umbraco-CMS
# AB 2439 -  Create API controller for uploading images for RTE

## Testing Notes
* Add RTE to a doctype
* Add Grid with a RTE editor allowed in a cell
* Drag a single image from your disk into a RTE
* Verify image uploaded into media section with correct name (matching filename)
* Verify the source code contents of the RTE does it have a `/media/` URL and a `data-udi` attribute set

* Repeat with two or more images dragged into editor
* Copy an image that is on your machine into your clipboard and paste it in such as word document or Google Doc etc

* Test with something thats NOT an image
* Update allowed image file extensions in Umbraco.config & try uploading

### Word & Google Docs
Currently cannot copy Word/Google Doc content with images in clipboard into RTE. You must select the image individually and copy/paste the image in.

## Considerations
Using native `localStorage` for this is this OK as think more widely supported now or do we need to use the AngularJS localStorage which I think may fallback to cookies or similar - thoughts?

## Quick Demo
![rte-media-upload](https://user-images.githubusercontent.com/1389894/64026540-de507c80-cb36-11e9-955d-fcef53ba61fb.gif)



Fixes [AB#2439](https://umbraco.visualstudio.com/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/2439)
